sprs/src/sparse/csmat.rs

```diff
@@ -549,17 +549,69 @@ impl<N, I: SpIndex, Iptr: SpIndex> CsMatI<N, I, Iptr> {
     }
 
     /// Append an outer dim to an existing matrix, compressing it in the process
-    pub fn append_outer(mut self, data: &[N]) -> Self
+    pub fn append_outer(self, data: &[N]) -> Self
     where
-        N: Clone + Num,
+        N: Clone + Zero,
```

Comment:
Will this (`Num` -> `Zero`) change break existing code though?

Reply:
As `Zero` is a subtrait I would say no https://docs.rs/num-traits/latest/num_traits/trait.Num.html